Lib/dataclasses.py

```diff
@@ -1184,17 +1184,21 @@ def _process_class(cls, init, repr, eq, order, unsafe_hash, frozen,
     return cls
 
 
-# _dataclass_getstate and _dataclass_setstate are needed for pickling frozen
-# classes with slots.  These could be slightly more performant if we generated
+# _dataclass_reduce and _dataclass_produce are needed because there is a
+# bug in pickle when there is a cycle in the object graph and the object
```

Comment:
i mean i would definitely consider it a bug, per https://github.com/python/cpython/issues/124937#issuecomment-2391891832 it was even reported 15 years ago at https://github.com/python/cpython/issues/45246 but never fixed. just because it was never fixed, does not make it not a bug.

but it's not that much difference to me, i can update the wording.

Reply:
430e441fff74294aa475f1e283df280df7cc799e

Reply:
thanks!  I was mostly trying to keep comments neutral (which is probably a fools errand on my part, as devs I expect we  opine about things in comments all the time).